packages/driver/cypress/integration/e2e/multi-domain/multi_domain_config_env_spec.ts

```diff
@@ -0,0 +1,260 @@
+['config', 'env'].forEach((fnName) => {
+  // @ts-ignore
+  describe(`multi-domain - Cypress.${fnName}()`, { experimentalSessionSupport: true, experimentalMultiDomain: true }, () => {
+    beforeEach(() => {
+      cy.visit('/fixtures/multi-domain.html')
+      cy.get('a[data-cy="multi-domain-secondary-link"]').click()
+    })
+
+    afterEach(() => {
+      // @ts-ignore
+      window.top.__cySkipValidateConfig = true
+    })
+
+    if (fnName === 'config') {
+      it(`throws if mutating read-only config values with Cypress.config()`, () => {
+        // @ts-ignore
+        window.top.__cySkipValidateConfig = false
+        cy.on('fail', (err) => {
+          expect(err.message).to.include('`Cypress.config()` cannot mutate option `chromeWebSecurity` because it is a read-only property.')
+        })
+
+        cy.switchToDomain('foobar.com', () => {
+          // @ts-ignore
+          Cypress.config('chromeWebSecurity', false)
+        })
+      })
+    }
+
+    context('serializable', () => {
+      it(`syncs initial Cypress.${fnName}() from the primary domain to the secondary (synchronously)`, () => {
+        Cypress[fnName]('foo', 'bar')
+        cy.switchToDomain('foobar.com', [fnName], ([fnName]) => {
+          const bar = Cypress[fnName]('foo')
+
+          expect(bar).to.equal('bar')
+        })
+      })
+
+      it(`syncs serializable values in the Cypress.${fnName}() again to the secondary even after spec bridge is created`, () => {
+        Cypress[fnName]('foo', 'baz')
+        cy.switchToDomain('foobar.com', [fnName], ([fnName]) => {
+          const baz = Cypress[fnName]('foo')
+
+          expect(baz).to.equal('baz')
+        })
+      })
+
+      it(`syncs serializable Cypress.${fnName}() values outwards from secondary (synchronously)`,
+        () => {
+          cy.switchToDomain('foobar.com', [fnName], ([fnName]) => {
+            Cypress[fnName]('bar', 'baz')
+          }).then(() => {
+            const baz = Cypress[fnName]('bar')
+
+            expect(baz).to.equal('baz')
+          })
+        })
+
+      it(`syncs serializable Cypress.${fnName}() values outwards from secondary even if the value is undefined`, () => {
+        Cypress[fnName]('foo', 'bar')
+
+        cy.switchToDomain('foobar.com', [fnName], ([fnName]) => {
+          Cypress[fnName]('foo', undefined)
+        }).then(() => {
+          expect(Cypress[fnName]('foo')).to.be.undefined
+        })
+      })
+
+      // FIXME: unskip this test once tail-end waiting is implemented
```

Comment:
I think what I assumed is that `switchToDomain` would wait for commands to execute and finish before the `then` is executed. Since the `then` gets executed after the `domain:fn` runs, I wonder if we need to sync the config twice regardless of how many commands in the queue, once after the `domain:fn` and after the commands in queue finish. Then this test would pass. What do you think?

Reply:
I'll see if we can unskip this once it's merged with my refactor.